Mathlib/Algebra/Algebra/Equiv.lean

```diff
@@ -811,3 +811,37 @@ def toAlgAut : G →* A ≃ₐ[R] A where
 end
 
 end MulSemiringAction
+
+namespace AlgEquiv
+section CompHom
+/--
+Define an equivalence of algebras in the following commutative diagram:
+A₁ <------> A₂
+/\         /\
+|           |
+|           |
+S --------> R
+in which `Algebra S A₂` is defined by `Algebra.compHom A₂ f`
+-/
+@[simps apply symm_apply toEquiv]
+def ofRingEquiv_compHom
+    {A₁: Type u} {A₂: Type v} {S: Type w} {R: Type u₁}
+    [Semiring A₁] [Semiring A₂]
+    [CommSemiring R] [CommSemiring S]
+    [Algebra R A₂] [Algebra S A₁]
+    (f: S →+* R) {equiv : A₁ ≃+* A₂}
+    (commute: ∀ x, (algebraMap R A₂) (f x) = equiv (algebraMap S A₁ x)) :
+    letI := Algebra.compHom A₂ f
+    A₁ ≃ₐ[S] A₂ :=
+  letI := Algebra.compHom A₂ f
+  {
+    equiv with
+    toFun := equiv,
+    invFun := equiv.symm,
+    commutes' := by
+        simp only [Algebra.compHom_algebraMap_eq, RingHom.coe_comp, Function.comp_apply]
+        exact (fun x => (commute x).symm)
+  }
```

Comment:
If you swap `commute`, then this `def` becomes a special case of `AlgEquiv.ofRingEquiv`. Given the awkward `letI`'s, I would be in favor of removing this `def` and just using `AlgEquiv.ofRingEquiv` whenever the need arises.
```suggestion
    (commute: ∀ x, equiv (algebraMap S A₁ x) = (algebraMap R A₂) (f x)) :
    letI := Algebra.compHom A₂ f
    A₁ ≃ₐ[S] A₂ :=
  letI := Algebra.compHom A₂ f
  AlgEquiv.ofRingEquiv commute
```